crates/astria-sequencer/src/accounts/action.rs

```diff
@@ -26,15 +27,36 @@ impl ActionHandler for TransferAction {
         &self,
         state: &S,
         from: Address,
+        fee_asset_id: &asset::Id,
     ) -> Result<()> {
-        let curr_balance = state
-            .get_account_balance(from)
+        let transfer_asset_id = self.asset_id;
+
+        let from_fee_balance = state
+            .get_account_balance(from, fee_asset_id)
             .await
-            .context("failed getting `from` account balance")?;
-        ensure!(
-            curr_balance >= self.amount + TRANSFER_FEE,
-            "insufficient funds"
-        );
+            .context("failed getting `from` account balance for fee payment")?;
+
+        // if fee asset is same as transfer asset, ensure accounts has enough funds
+        // to cover both the fee and the amount transferred
+        if fee_asset_id == &transfer_asset_id {
+            ensure!(
+                from_fee_balance >= self.amount + TRANSFER_FEE,
```

Comment:
Can we use `let Some(from_fee_balance) = self.amount.checked_add(TRANSFER_FEE) else {};` here? These things are *usually* fine, but I'd like us to be extra cautious here because the worst case scenario (however unlikely) is bad.